vault/request_handling.go

```diff
@@ -1406,7 +1407,7 @@ func (c *Core) handleLoginRequest(ctx context.Context, req *logical.Request) (re
 				return nil, nil, err
 			}
 		}
-		return nil, nil, resp.Error()
+		return resp, nil, routeErr
```

Comment:
Just gave this a test with a different auth method (userpass) to see if there are any undesirable side effects. I think this fixes other issues (good thing!). Interesting that userpass auth with an incorrect password returns a 500 before this change. Makes me wonder how long it's behaved like that 🤔 With the changes in this PR, both login attempts below return a 400.

```
$ vault auth enable userpass

$ vault write auth/userpass/users/test \
    password=password

$ vault login -method=userpass username=testt password=password
Error authenticating: Error making API request.

URL: PUT http://localhost:8200/v1/auth/userpass/login/testt
Code: 400. Errors:

* invalid username or password
$ vault login -method=userpass username=test password=passwordd
Error authenticating: Error making API request.

URL: PUT http://localhost:8200/v1/auth/userpass/login/test
Code: 500. Errors:

* invalid username or password
```

Reply:
Thanks for testing the userpass auth flow!

Reply:
If the error is non-nil in this case (and scoped to be `logical.ErrInvalidCredentials` from the check above), is there a reason to return `resp` rather than nil?

Reply:
I believe so that a 204 No Content isn't returned (reported in the [GH issue](https://github.com/hashicorp/vault/issues/20923)) and the response has the `errUserBindFailed` contents included. @raymonstah can confirm this.

Reply:
I see what's going -- the `resp`'s Error is what fixes the 204 since the actual content is coming from there, and not the `error`'s string value.